src/irmin-pack/unix/mapping_file.ml

```diff
@@ -365,95 +361,52 @@ module Make (Errs : Io_errors.S with module Io = Io.Unix) = struct
     Int_mmap.close file0;
     Io.Unix.unlink path0 |> ignore;
 
-    (* Create [file2] *)
-    let file2_ref = ref None in
-    let auto_flush_callback () =
-      match !file2_ref with
-      | None -> assert false
-      | Some x -> Ao.flush x |> Errs.raise_if_error
-    in
-    let* file2 =
-      Ao.create_rw ~path:path2 ~overwrite:true ~auto_flush_threshold:1_000_000
-        ~auto_flush_callback
```

Comment:
Except for the mmaps, all of irmin-pack does IO using `io.ml`. This centralization has advantages. 

Would you consider switching back to the Append_only module here?

Reply:
In the interest of getting some code merged as soon as possible, I'll keep it as is for the time being. This is only for the mmap'ed mapping file, which already is treated "specially" (i.e., not using the Io routines), so hopefully we can keep the mmap as a special case, and use Io for everything else.

Reply:
Let's do that